oak/client/application_client.h

```diff
@@ -97,22 +97,13 @@ class ApplicationClient {
     return grpc::CreateChannel(addr, composite_credentials);
   }
 
-  // Returns a gRPC Channel initialized with TLS channel credentials and a private authorization
-  // bearer token.
-  //
-  // See `private_authorization_bearer_token_call_credentials`.
-  static std::shared_ptr<grpc::Channel> CreateTlsChannel(std::string addr) {
-    return CreateChannel(addr, GetTlsChannelCredentials(),
-                         private_authorization_bearer_token_call_credentials());
-  }
-
-  // Returns a gRPC Channel initialized with Tls channel credentials and a non-default PEM-encoded
-  // CA root certificate, and a private authorization bearer token.
-  //
-  // See `private_authorization_bearer_token_call_credentials`.
-  static std::shared_ptr<grpc::Channel> CreateTlsChannel(std::string addr, std::string ca_cert) {
-    return CreateChannel(addr, GetTlsChannelCredentials(ca_cert),
-                         private_authorization_bearer_token_call_credentials());
+  // Returns a grpc Channel connecting to the specified address initialised a fixed Oak Label.
+  static std::shared_ptr<grpc::Channel> CreateChannel(
+      std::string addr, std::shared_ptr<grpc::ChannelCredentials> channel_credentials,
+      oak::label::Label label) {
+    auto call_credentials =
+        grpc::MetadataCredentialsFromPlugin(absl::make_unique<LabelMetadata>(label));
+    return CreateChannel(addr, channel_credentials, call_credentials);
```

Comment:
What about the non-C++ client examples (e.g. `translator/client/translator.go`, `hello_world/client/nodejs/`) ? I think they could do with at least a `TODO` 

[TDD: TODO-driven development :-) ]

Reply:
Oh you are right, are those not being run at all currently?

Reply:
`scripts/run_examples` should run at least the `translator` example, so if cloudbuild is passing it's worth checking what's happening there.

Reply:
It looks like it's just logging an error but nothing notices:

https://github.com/project-oak/oak/blob/fedfba2ef22194edc9b5fbee3c5f37ec707a566e/examples/translator/client/translator.go#L36-L40

If I change that to `Fatalf` then it does fail, and perhaps that's what we should do? I'll raise a separate issue.

Reply:
That won't quite work as-is, because the translator returns `NotFound` for requests that it doesn't have the vocabulary for &ndash; it will need something that depends on the error code.

That does make me think more generally though:  this PR is adding new label-checking behaviour, so it should also include tests for that new checking behaviour.

Maybe a couple of new test cases in `examples/abitest/client/grpctest.cc` that send in requests with 0 or 2 labels, and expect a failure?  (It will need a little bit of a re-org, because it can't share the usual `OakABITestService::Stub*`)

Reply:
Yeah good point, I thought about that but then realised that indeed the shared stub would not allow that. I've created https://github.com/project-oak/oak/issues/1072 to track, as I think I may do a bit of trial and error to figure out the best approach. I'll work on it right after this PR.